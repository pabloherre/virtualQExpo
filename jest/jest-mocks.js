import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.requireActual('./mocks');
jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
jest.mock('expo-barcode-scanner');
jest.mock('expo-location');
jest.mock('react-native-maps');
