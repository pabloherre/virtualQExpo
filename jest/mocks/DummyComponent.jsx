import React, { Component } from 'react';
import { View } from 'react-native';

export default class DummyComponent extends Component {
  render() {
    return <View></View>;
  }
}
