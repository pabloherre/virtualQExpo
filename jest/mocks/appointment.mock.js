export const appointment = {
  id: 1,
  date: 'Today',
  business: 'Pharmacy',
  appointmentNumber: '234234',
  appointmentCountdown: 23,
  latitude: -32.8913568,
  longitude: -68.8567443,
  address: 'Av. Cabildo 2035',
  city: 'Belgrano, CABA',
  email: 'somename@randomemail.com'
};
