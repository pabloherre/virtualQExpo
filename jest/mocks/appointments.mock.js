export const appointments = [
  {
    id: 1,
    date: 'Today',
    business: 'Pharmacy',
    appointmentNumber: '234234',
    appointmentCountdown: 23,
    latitude: -32.8913568,
    longitude: -68.8567443
  },
  {
    id: 2,
    date: 'Today',
    business: 'Supermarket',
    appointmentNumber: '2B4354',
    appointmentCountdown: 322,
    latitude: -32.8913568,
    longitude: -68.8567443
  },
  {
    id: 3,
    date: 'Tomorrow',
    business: 'Cinema',
    appointmentNumber: '234354',
    appointmentCountdown: 322,
    latitude: -32.8913568,
    longitude: -68.8567443
  },
  {
    id: 4,
    date: 'Tomorrow',
    business: 'Pool',
    appointmentNumber: '234354',
    appointmentCountdown: 322,
    latitude: -32.8913568,
    longitude: -68.8567443
  }
];
