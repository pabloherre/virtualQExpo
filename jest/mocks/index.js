export * from './appointment.mock';
export * from './appointments.mock';

