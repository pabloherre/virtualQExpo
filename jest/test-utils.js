import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);
function connectedRender(ui, initialState) {
  const store = mockStore(initialState);
  return renderer.create(<Provider store={store}>{ui}</Provider>);
}

function connectedShallow(ui, initialState) {
  const store = mockStore(initialState);
  return shallow(<Provider store={store}>{ui}</Provider>);
}

function connectedMount(ui, initialState, numberOfHoc) {
  const store = mockStore(initialState);
  return mount(<Provider store={store}>{ui}</Provider>);
}

// override connectedRender method
export { connectedRender, connectedShallow, connectedMount };
