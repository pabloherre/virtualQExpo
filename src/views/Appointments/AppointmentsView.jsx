import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ActivityIndicator, FlatList, TouchableOpacity, View } from 'react-native';
import styles from './AppointmentsView.styles';
import AppointmentCard from '../../modules/appointments/components/AppointmentCard';
import { compose } from 'redux';
import checkAuthentication from '../../modules/auth/components/CheckAuthentication';
import { safeArea } from '../../styles/common.styles';
import { Entypo } from '@expo/vector-icons';
import RoundedIcon from '../../common/icons/RoundedIcon';
import { colors } from '../../../theme';
import { AppointmentDefinition } from '../../modules/appointments/appointments.definitions';
import AppointmentService from '../../services/appointment/Appointment.service';
import { setTestInfo } from '../../utils/test.utils';

export class AppointmentsView extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    AppointmentService.findAppointments();
  }

  handleOnPressCard = appointment => {
    this.props.navigation.navigate('AppointmentDetails', { appointment });
  };

  render() {
    const { appointments, isLoading } = this.props;

    return (
      <View style={safeArea}>
        {isLoading ? (
          <ActivityIndicator size="large" color={colors.primary} />
        ) : (
          <FlatList
            data={appointments}
            {...setTestInfo('testListAppointmentsList')}
            renderItem={({ item }) => (
              <AppointmentCard
                appointment={item}
                onPressCard={appointment => this.handleOnPressCard(appointment)}
                {...setTestInfo('testTouchableAppointmentCard')}
              />
            )}
            keyExtractor={item => item._id}
          />
        )}
        <View style={styles.buttonContainer}>
          <View style={styles.addButton}>
            <RoundedIcon size={64} bkColor={colors.primary} shadow={true}>
              <TouchableOpacity
                onPress={() => this.props.navigation.navigate('AppointmentNew')}
                {...setTestInfo('testTouchableNewAppointmentButton')}
              >
                <Entypo name="plus" size={36} color="white" />
              </TouchableOpacity>
            </RoundedIcon>
          </View>
        </View>
      </View>
    );
  }
}

AppointmentsView.propTypes = {
  colors: PropTypes.shape({
    primary: PropTypes.string
  }),
  appointments: PropTypes.arrayOf(PropTypes.shape(AppointmentDefinition))
};

AppointmentsView.defaultProps = {
  colors: colors
};

const mapStateToProps = state => {
  return {
    appointments: state.appointment.data,
    isLoading: state.appointment.loading
  };
};

export default compose(checkAuthentication, connect(mapStateToProps, null))(AppointmentsView);
