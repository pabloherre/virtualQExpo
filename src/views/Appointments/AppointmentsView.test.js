import React from 'react';
import { mount, shallow } from 'enzyme';

import { AppointmentsView } from './AppointmentsView';
import { appointments } from '../../../jest/mocks';
import { connectedShallow, connectedMount } from '../../../jest/test-utils';

const mockProps = {
  appointments,
  navigation: {
    navigate: jest.fn()
  }
};

describe('<AppointmentsView />', () => {
  afterEach(() => {
    mockProps.navigation.navigate.mockClear();
  });
  it('should render correctly', async () => {
    const wrapper = shallow(<AppointmentsView />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should pass the appointments to the flatList component', async () => {
    const wrapper = shallow(<AppointmentsView {...mockProps} />);

    expect(wrapper.find('FlatList').shallow().instance().props.data.length).toBe(5);
  });

  it('should redirect with the appointment when Appointment Card callback is called', async () => {
    const wrapper = shallow(<AppointmentsView {...mockProps} />);

    wrapper.instance().handleOnPressCard();

    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled();
  });

  it('should have an add button that redirects you when pressed', async () => {
    const wrapper = shallow(<AppointmentsView {...mockProps} />);

    wrapper.find('RoundedIcon').childAt(0).props().onPress();

    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled();
  });
});
