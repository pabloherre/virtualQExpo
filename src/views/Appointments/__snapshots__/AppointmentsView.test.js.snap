// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentsView /> should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingHorizontal": 40,
      "paddingVertical": 20,
    }
  }
>
  <FlatList
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <View
    style={
      Object {
        "bottom": 20,
        "left": 40,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
        }
      }
    >
      <RoundedIcon
        bkColor="#f77027"
        shadow={true}
        size={64}
      >
        <ForwardRef
          onPress={[Function]}
        >
          <Icon
            allowFontScaling={false}
            color="white"
            name="plus"
            size={36}
          />
        </ForwardRef>
      </RoundedIcon>
    </View>
  </View>
</View>
`;
